### DISCUSSION TITLE
Question about the length of the input sequence

### DISCUSSION THREAD

**Discussion by Yang-J-LIN** [facebookresearch/esm#76] [Q&A] [answered]
Hi,

Thanks for the great work.

A few questions came out when I was using the ESM model. And I come here for your help.

The first question is about the limitation of the input length. It seems that the maximum sequence length is 1024. I am wondering, during the test, whether there is a possibility to adapt the model to suit the sequence longer than 1024 AA.

The second question is about the generalization performance on short sequences. The model is trained on UniProt dataset, who contains sequences most of which have lengths of hundreds. Can ESM extract information from very short sequences? For example, a helix or a sheet, whose lengths often range from 5 to 50.

Thanks for your help in advance.

**Comment by tomsercu** [comment, marked as answer]
1. See the discussion in #21 . There's also some discussion of the different positional embedding types in #49 .
2. Probably the model will do something reasonable, but it's an open/experimental question. I think your expectation about what is reasonable for the model to predict should be upper bounded by what properties really physically exist for a fragment in isolation, without full protein context. I suspect most things don't make sense for fragments in isolation, like predicting tertiary structure, function, stability etc..

**Reply by Yang-J-LIN** [reply]
Thanks for the reply. I have tested it with the short sequences and the result is bad.
